src/librustc_expand/mbe/macro_rules.rs

```diff
@@ -286,7 +317,9 @@ fn generic_extension<'cx>(
                 mbe::TokenTree::Delimited(_, ref delim) => &delim.tts[..],
                 _ => continue,
             };
-            match parse_tt(cx, lhs_tt, arg.clone()) {
+            let base_parser =
+                base_parser_from_cx(&cx.current_expansion, &cx.parse_sess, arg.clone());
```

Comment:
This is an error-only path, so what we are doing here doesn't make much difference in terms of performance, but it looks like it would be simpler to reuse the `base_parser` and `parser` created above here instead of dropping them.

Reply:
This isn't valid. Although the text of the two `base_parser` assignments is identical, the second one has a different `arg` because of the `if let Some((_arg, comma_span)) = arg.add_comma()` above. And when I tried it, I get these failures:
```
failures:
    [ui] ui/macros/assert-trailing-junk.rs
    [ui] ui/macros/missing-comma.rs
```